pkg/kn/commands/types.go

```diff
@@ -38,12 +40,13 @@ type Config struct {
 	LookupPlugins bool
 }
 
-// Parameters for creating commands. Useful for inserting mocks for testing.
+// KnParams for creating commands. Useful for inserting mocks for testing.
 type KnParams struct {
-	Output       io.Writer
-	KubeCfgPath  string
-	ClientConfig clientcmd.ClientConfig
-	NewClient    func(namespace string) (serving_kn_v1alpha1.KnServingClient, error)
+	Output           io.Writer
+	KubeCfgPath      string
+	ClientConfig     clientcmd.ClientConfig
+	NewClient        func(namespace string) (serving_kn_v1alpha1.KnServingClient, error)
+	NewSourcesClient func(namespace string) (sources_kn_v1alpha1.KnSourcesClient, error)
 
 	// General global options
 	LogHTTP bool
```

Comment:
Interesting. Where is this being used?

Reply:
When we begin to add Eventing functions to client, we may hope to have all methods to interact with Eventing server APIs in a single file. Similar as [pkg/serving/v1alpha1/client.go](https://github.com/knative/client/blob/master/pkg/serving/v1alpha1/client.go), I created pkg/sources/v1alpha1/client.go correspondingly. NewSourcesClient will be used to create such a KnSourcesClient defined in pkg/sources/v1alpha1/client.go. It will be invoked in every command who needs to interact with eventing sources api.